src/test/regress/expected/single_node.out

```diff
@@ -2163,24 +2163,24 @@ NOTICE:  executing the command locally: WITH cte_1 AS (UPDATE single_node.anothe
 SELECT count(*) from another_schema_table;
-ERROR:  could not establish any connections to the node localhost:xxxxx when local execution is also disabled.
-HINT:  Enable local execution via SET citus.enable_local_execution TO true;
+ERROR:  could not establish any connections to the node localhost:xxxxx due to high number connections.
+HINT:  Consider enabling local execution using "SET citus.enable_local_execution TO true;" (if not enabled already) or setting max_connections to a higher value
 UPDATE another_schema_table SET b = b;
-ERROR:  could not establish any connections to the node localhost:xxxxx when local execution is also disabled.
-HINT:  Enable local execution via SET citus.enable_local_execution TO true;
+ERROR:  could not establish any connections to the node localhost:xxxxx due to high number connections.
+HINT:  Consider enabling local execution using "SET citus.enable_local_execution TO true;" (if not enabled already) or setting max_connections to a higher value
 INSERT INTO another_schema_table SELECT * FROM another_schema_table;
-ERROR:  could not establish any connections to the node localhost:xxxxx when local execution is also disabled.
-HINT:  Enable local execution via SET citus.enable_local_execution TO true;
+ERROR:  could not establish any connections to the node localhost:xxxxx due to high number connections.
+HINT:  Consider enabling local execution using "SET citus.enable_local_execution TO true;" (if not enabled already) or setting max_connections to a higher value
 INSERT INTO another_schema_table SELECT b::int, a::int FROM another_schema_table;
-ERROR:  could not establish any connections to the node localhost:xxxxx when local execution is also disabled.
-HINT:  Enable local execution via SET citus.enable_local_execution TO true;
+ERROR:  could not establish any connections to the node localhost:xxxxx due to high number connections.
+HINT:  Consider enabling local execution using "SET citus.enable_local_execution TO true;" (if not enabled already) or setting max_connections to a higher value
 WITH cte_1 AS (SELECT * FROM another_schema_table LIMIT 1000)
 	SELECT count(*) FROM cte_1;
-ERROR:  could not establish any connections to the node localhost:xxxxx when local execution is also disabled.
-HINT:  Enable local execution via SET citus.enable_local_execution TO true;
+ERROR:  could not establish any connections to the node localhost:xxxxx due to high number connections.
+HINT:  Consider enabling local execution using "SET citus.enable_local_execution TO true;" (if not enabled already) or setting max_connections to a higher value
 INSERT INTO another_schema_table VALUES (1,1), (2,2), (3,3), (4,4), (5,5),(6,6),(7,7);
-ERROR:  could not establish any connections to the node localhost:xxxxx when local execution is also disabled.
-HINT:  Enable local execution via SET citus.enable_local_execution TO true;
+ERROR:  could not establish any connections to the node localhost:xxxxx due to high number connections.
```

Comment:
what happens if you execute a command that does not support local execution? it should give another error?

Reply:
True, but given that at that point we cannot know whether local execution is supported or not (or was disabled by user), I thought that emphasizing `(if not enabled already)` would be nice & enough. What do you think ?

Reply:
I meant, shouldn't we add the test you have on the PR description to here as well

Reply:
Let me try to add an isolation test for that

Reply:
added a plain regression test instead